### DISCUSSION TITLE
question about the Bluetooth protocol for connecting and transferring information to cstimer

### DISCUSSION THREAD

**Discussion by raphchta** [cs0x7f/cstimer#304] [Q&A] [answered]
i try create my timer.

My problem is that csTimer doesn't recognise my timer and I don't know how to transfer my data to csTimer.

My timer is running on an esp32 and is not displayed when I scan the network with my browser. I would also like to know what information I can transfer and how to do it.
thank you for your help

**Comment by raphchta** [comment, marked as answer]
i forgot to say that i want to connect via bluetooth, sorry

**Reply by afedotov** [reply]
Obviously if you implement your own protocol, csTimer won't understand it.

Otherwise you can mimic GAN timer protocol. All information you need like required device name, service and characteristic UUIDs, event format and their behavior you can get from source code here - [src/js/gantimer.js](https://github.com/cs0x7f/cstimer/blob/master/src/js/gantimer.js)

Alternatively you can use UART and mimic StackMat protocol, specs available on the internet, for example - [Protocol.md](https://github.com/PaddyCo/StackmatTimer/blob/main/Protocol.md)

**Reply by raphchta** [reply]
where can I find the event code because I don't have it all in the code?

**Reply by afedotov** [reply]
Can't understand what exactly are you need. All is in the code.
If you have difficulties to read JavaScript, I put protocol specs here - [GAN Smart Timer bluetooth protocol](https://gist.github.com/afedotov/a025fa5796c9c727b04cf98b293a02f6)
